### ISSUE TITLE
Hangul module: grouping and composition rules (variant assignment) 

### ISSUE THREAD

**Issue by GaetanBaehr** [BlackFoundryCom/black-robo-cjk#45]

**Comment by aaronbell** [comment]
The goal of automation is to enable a designer to establish relationships between jamo, assign jamo variants to specific combinations, and auto-generate all the desired syllables in the Korean set (KSX 1001 or the full 11,000). 

This mechanism relies on three parts to be successful:

1) Jamo definitions – this lists the variants for each jamo that will be used by the system for generating the syllables.

2) Jamo groupings – this lists the jamo that we will treat in the same way (as in, will use the same jamo variant) when it appears in a syllable combination. 

3) Assignment – this lists out all of the possible combinations of jamo groups, and then indicates which of the jamo variants to use for that syllable combination. 

With the above in place, we will be able to automatically generate a syllable with the appropriate variant for each jamo in the syllable.

Looking forward to seeing how the implementation develops!

**Comment by aaronbell** [comment]
Something I've been thinking about recently is if it would be possible to do two separate 'jamo groupings'. 

For example, the vertical vowels (ex:ㅣㅏㅓㅕㅑ) need to adapt to the final vowels in different ways than the horizontal vowels (ex: ㅡㅜㅠㅗㅛ). So where in a vertical setting, a final jamo of ㅇand ㅁ are treated differently, we might be able to treat them the same way in a horizontal setting. 

If we use the same jamo groupings for both, then one has to create rules to account for the different use cases, even where the same variant is being used across various jamo assignments.

**Comment by JeremieHornus** [comment]
This would mean one glyph can belong to two (or more) groups, then the groups usage is context-dependent, right?

**Comment by aaronbell** [comment]
Yes. You'd have a 'horizontal' group and a 'vertical' group that would be exclusive based on the vowel jamo used.

**Comment by aaronbell** [comment]
Hm. Let me think on this more. The idea might be an byproduct of the process I used rather than something that would be truly beneficial.
